AirCasting/MainTabBarView.swift

```diff
@@ -54,14 +55,17 @@ struct MainTabBarView: View {
             
         }
         .onAppCameToForeground {
-            measurementUpdatingService.updateAllSessionsMeasurements()
+            measurementsDownloadingInProgress = true
+            measurementUpdatingService.updateAllSessionsMeasurements() { measurementsDownloadingInProgress = false }
```

Comment:
doesn't Xcode warn about this state being set on a non-main thread? Because the completion here is called from a background queue